### PR TITLE
ImageUtil

### PR DESCRIPTION
**Pull request by floscher**
This changes the import of images in the following way:
* the import code is in the new class `ImportUtil`
* directories are imported recursively instead of only images in the top level
* both imports of complete sequences as well as single images accept JPG **and** PNG
* the file type of the images is no longer determined by the file extension, but by the [magic bytes](https://en.wikipedia.org/wiki/List_of_file_signatures)

### CODE REVIEW

File: src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      if (meta instanceof JpegImageMetadata) {
+        final JpegImageMetadata jpegMeta = (JpegImageMetadata) meta;
+        latRef = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_LATITUDE_REF);
+        lonRef = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_LONGITUDE_REF);
+        lat = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_LATITUDE);
+        lon = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_LONGITUDE);
+        gpsDir = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_IMG_DIRECTION);
+        dateTime = getTiffFieldValue(jpegMeta, ExifTagConstants.EXIF_TAG_DATE_TIME_ORIGINAL);
+      }
+    } catch (ImageReadException e) {
```
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java]: Codacy Issue found: [Avoid empty catch blocks](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1776583746/issues/source?bid=3023888&fileBranchId=3066982#l93)

File: src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    }
+    return null;
+  }
+
+  private static class ImageFileFilter extends FileFilter {
+    private static final byte[] JFIF_MAGIC = new byte[]{-1 /*0xFF*/, -40 /*0xD8*/};
+    private static final byte[] PNG_MAGIC = new byte[]{
+        -119 /*0x89*/, 80 /*0x50*/, 78 /*0x4E*/, 71 /*0x47*/, 13 /*0x0D*/, 10 /*0x0A*/, 26 /*0x1A*/, 10 /*0x0A*/
+    };
+    private byte[] magic = new byte[Math.max(JFIF_MAGIC.length, PNG_MAGIC.length)];
```
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java]: Codacy Issue found: [Private field 'magic' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1776583746/issues/source?bid=3023888&fileBranchId=3066982#l136)

File: src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      if (meta instanceof JpegImageMetadata) {
+        final JpegImageMetadata jpegMeta = (JpegImageMetadata) meta;
+        latRef = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_LATITUDE_REF);
+        lonRef = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_LONGITUDE_REF);
+        lat = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_LATITUDE);
+        lon = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_LONGITUDE);
+        gpsDir = getTiffFieldValue(jpegMeta, GpsTagConstants.GPS_TAG_GPS_IMG_DIRECTION);
+        dateTime = getTiffFieldValue(jpegMeta, ExifTagConstants.EXIF_TAG_DATE_TIME_ORIGINAL);
+      }
+    } catch (ImageReadException e) {
```
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java]: Codacy Issue found: [Avoid empty catch blocks](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1776977857/issues/source?bid=3023888&fileBranchId=3066982#l93)

File: src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  private static class ImageFileFilter extends FileFilter {
+    private static final byte[] JFIF_MAGIC = new byte[]{-1 /*0xFF*/, -40 /*0xD8*/};
+    private static final byte[] PNG_MAGIC = new byte[]{
+        -119 /*0x89*/, 80 /*0x50*/, 78 /*0x4E*/, 71 /*0x47*/, 13 /*0x0D*/, 10 /*0x0A*/, 26 /*0x1A*/, 10 /*0x0A*/
+    };
+    private final byte[] magic = new byte[Math.max(JFIF_MAGIC.length, PNG_MAGIC.length)];
+
+    ImageFileFilter() { }
```
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1776977857/issues/source?bid=3023888&fileBranchId=3066982#l138)

File: src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java (hunk 3 above)
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java]: Codacy Issue found: [Avoid empty catch blocks](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1804015725/issues/source?bid=3023888&fileBranchId=3066982#l93)

File: src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java (hunk 4 above)
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1804015725/issues/source?bid=3023888&fileBranchId=3066982#l138)

File: src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java (hunk 3 above)
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java]: Codacy Issue found: [Avoid empty catch blocks](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1804015725/issues/source?bid=3023888&fileBranchId=3066982#l93)

File: src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java (hunk 4 above)
Comment (floscher) [inline src/org/openstreetmap/josm/plugins/mapillary/utils/ImageUtil.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/floscher/josm-mapillary-plugin/file/1804015725/issues/source?bid=3023888&fileBranchId=3066982#l138)

**Comment by nokutu** [comment]
Sorry, I forgot about the this haha. The point I didn't support png at first was because they didn't store gps data, so they would all appear in the center, which would be kind of a mess. But anyways, it would probably be a good idea to allow the user to get that mess

**Pull request merged**